var/spack/repos/builtin/packages/silo/package.py

```diff
@@ -45,8 +47,9 @@ class Silo(AutotoolsPackage):
     def flag_handler(self, name, flags):
         spec = self.spec
         if name == 'ldflags':
-            if spec['hdf5'].satisfies('~shared'):
-                flags.append('-ldl')
+            if spec.satisfies('+hdf5'):
```

Comment:
Variant checks normally appear as:

```suggestion
            if '+hdf5' in spec:
```